### ISSUE TITLE
withCapture(x <- 1L) drops suffix L in non-interactive mode

### ISSUE THREAD

**Issue by HenrikBengtsson** [HenrikBengtsson/R.utils#58]
Calling `withCapture(x <- 1L)` in non-interactive mode drops the `L` suffix from integer `1L`, e.g.
```sh
$ Rscript -e 'sQuote(R.utils::withCapture(x <- 1L))'
[1] "‘> x <- 1\n’"
```
and
```sh
$ echo "sQuote(R.utils::withCapture(x <- 1L))" | R --quiet --vanilla
> sQuote(R.utils::withCapture(x <- 1L))
[1] "‘> x <- 1\n’"
> 
```

It _does_ work in interactive mode, e.g.
```r
> sQuote(R.utils::withCapture(x <- 1L))
[1] "‘> x <- 1L\n’"
```

**Comment by HenrikBengtsson** [comment]
More troubleshooting: it turns out that this originates from how `source(..., echo=TRUE)` works in interactive and non-interactive mode, e.g.

```sh
$ echo "x <- 10L" > code.R
$ Rscript -e "source('code.R', echo=TRUE)"
> x <- 10

$ R --vanilla --quiet
> source('code.R', echo=TRUE)
> x <- 10L
```

**Comment by HenrikBengtsson** [comment]
Aha, the difference is in options `keep.source`, which is defined as `options(keep.source=interactive())`, e.g.

```sh
$ Rscript -e "source('code.R', echo=TRUE)"
> x <- 10

$ Rscript -e "source('code.R', echo=TRUE, keep.source=TRUE)"
> x <- 10L

$ Rscript -e "source('code.R', echo=TRUE, keep.source=FALSE)"
> x <- 10
```

So, `R.utils::withCapture()` should make sure to always call `source()` with `keep.source=TRUE`.

**Comment by HenrikBengtsson** [comment]
Fixed in commit 84428d5c.
